File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/ModelWithoutDefaultLoadersTests.java
package org.cqframework.cql.cql2elm;

import static kotlinx.io.CoreKt.buffered;
import static kotlinx.io.JvmCoreKt.asSource;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.io.IOException;
import org.hl7.cql.model.ModelInfoProvider;
import org.hl7.cql.model.NamespaceManager;
import org.hl7.elm.r1.Library;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class ModelWithoutDefaultLoadersTests {
    private static ModelManager modelManager;
    private static ModelInfoProvider modelInfoProvider;

    @BeforeAll
    static void setup() {
        modelManager = new ModelManager(new NamespaceManager(), false);
        modelInfoProvider = new TestModelInfoProvider();
        // modelManager.getModelInfoLoader().registerSystemModelInfoProvider();
        modelManager.getModelInfoLoader().registerModelInfoProvider(modelInfoProvider);
    }

    @AfterAll
    static void tearDown() {
        modelManager.getModelInfoLoader().unregisterModelInfoProvider(modelInfoProvider);
    }

    @Test
    void modelInfo() {
        CqlTranslator translator = null;
        try {
            translator = CqlTranslator.fromSource(
                    buffered(asSource(ModelWithoutDefaultLoadersTests.class.getResourceAsStream("ModelTests/ModelTest.cql"))),
                    new LibraryManager(modelManager));
            Library library = translator.toELM();
            assertThat(translator.getErrors().size(), is(0));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Src/java/cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/ModelWithoutDefaultLoadersTests.kt
package org.cqframework.cql.cql2elm

import java.io.IOException
import kotlinx.io.asSource
import kotlinx.io.buffered
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hl7.cql.model.ModelInfoProvider
import org.hl7.cql.model.NamespaceManager
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

internal class ModelWithoutDefaultLoadersTests {
    @Test
    fun modelInfo() {
        var translator: CqlTranslator?
        try {
            translator =
                CqlTranslator.fromSource(
                    ModelWithoutDefaultLoadersTests::class
                        .java
                        .getResourceAsStream("ModelTests/ModelTest.cql")!!
                        .asSource()
                        .buffered(),
                    LibraryManager(modelManager!!),
                )
            translator.toELM()
            assertThat(translator.errors.size, `is`(0))
        } catch (e: IOException) {
            @Suppress("PrintStackTrace") e.printStackTrace()
        }
    }

    companion object {
        private var modelManager: ModelManager? = null
        private var modelInfoProvider: ModelInfoProvider? = null

        @JvmStatic
        @BeforeAll
        fun setup() {
            modelManager = ModelManager(NamespaceManager(), false)
            modelInfoProvider = TestModelInfoProvider()
            // modelManager.getModelInfoLoader().registerSystemModelInfoProvider();
            modelManager!!.modelInfoLoader.registerModelInfoProvider(modelInfoProvider!!)
        }

        @JvmStatic
        @AfterAll
        fun tearDown() {
            modelManager!!.modelInfoLoader.unregisterModelInfoProvider(modelInfoProvider!!)
        }
    }
}
